### PR TITLE
chore: Updates atlas-github-action for contract tests

### PR DESCRIPTION
**Pull request by maastha**
## Proposed changes

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)
- [ ] This change requires a documentation update
- [ ] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Manual QA performed:

- [ ] cfn invoke for each of CRUDL/cfn test
- [ ] Updated resource in  [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples)
- [ ] Published to AWS private registry
- [ ] Used the template in [example](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/examples) to create and update a stack in AWS
- [ ] Deleted stack to ensure resources are deleted
- [ ] Created multiple resources in same stack
- [ ] Validated in Atlas UI
- [ ] Included screenshots

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [ ] I have added any necessary documentation (if appropriate)
- [ ] I have run `make fmt` and formatted my code
- [ ] For CFN Resources: I have released by changes in the private registry and proved by change
  works in Atlas

### CODE REVIEW

File: .github/workflows/contract-testing.yaml [hunk 1]
```diff
@@ -69,11 +69,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: setup Atlas CLI
-        uses: andreaangiolillo/atlas-cli-github-action@main
-        with:
-          public-key: ${{ secrets.CLOUD_DEV_PUBLIC_KEY }}
-          private-key: ${{ secrets.CLOUD_DEV_PRIVATE_KEY }}
-          org-id: ${{ secrets.CLOUD_DEV_ORG_ID }}
+        uses: mongodb/atlas-github-action@v0.2.0
```
Comment (lantoli) [inline .github/workflows/contract-testing.yaml:72]: i think we decided to use SHAs in actions

**Comment by andreaangiolillo** [comment]
Could you check if we also have to update cdk?

File: .github/workflows/contract-testing.yaml (hunk 1 above)
Comment (maastha) [reply .github/workflows/contract-testing.yaml]: Thanks @lantoli great catch, updated to use SHA

**Review by lantoli — APPROVED** [review]

File: .github/workflows/contract-testing.yaml (hunk 1 above)
Comment (lantoli) [reply .github/workflows/contract-testing.yaml]: thx for addressing this

**Comment by maastha** [comment]
> Could you check if we also have to update cdk?

checked, CDK doesn't seem to be using atlas cLI action so we should be good there

**Pull request merged**